tests/lib/rules/html-indent.js

```diff
@@ -9,18 +9,69 @@
 // Requirements
 // ------------------------------------------------------------------------------
 
-const rule = require('../../../lib/rules/html-indent')
+const fs = require('fs')
+const path = require('path')
 const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/html-indent')
 
 // ------------------------------------------------------------------------------
 // Helpers
 // ------------------------------------------------------------------------------
 
+const FIXTURE_ROOT = path.resolve(__dirname, '../../fixtures/html-indent/')
+
+/**
+ * Load test patterns from fixtures.
+ * @param {object[]} additionalValid The array of additional valid patterns.
+ * @param {object[]} additionalInvalid The array of additional invalid patterns.
+ * @returns {object} The loaded patterns.
+ */
+function loadPatterns (additionalValid, additionalInvalid) {
+  const valid = fs
+    .readdirSync(FIXTURE_ROOT)
+    .map(filename => {
+      const code0 = fs.readFileSync(path.join(FIXTURE_ROOT, filename), 'utf8')
+      const code = code0.replace(/^<!--(.+?)-->/, `<!--${filename}-->`)
+      const baseObj = JSON.parse(/^<!--(.+?)-->/.exec(code0)[1])
+      return Object.assign(baseObj, { code, filename })
+    })
+  const invalid = valid
```

Comment:
It's hard to read and know which fixture is good, or not. You're detecting them here, but I think it would be muuuch easier to just add another comment or put valid fixtures in `valid` directory and so on. Then you wouldn't need to do any confusing magic here. Or am I'm missing something? (It's very possible :D)

Reply:
I'll add comments about what this does.

This addresses the code in fixtures as valid.
And this creates an invalid case from every valid case -- it removes all indentation and checks whether the rule restores all indentation or not.

Reply:
So if there are ignored lines, we can't use fixtures. And it's the reason that some tests exist on outside of fixtures.

Reply:
I added comments.

To realize `invalid` sub-directory, it needs a DSL to specify expected indentation for each line. Indeed it would be useful, but I avoided it for my time.

Reply:
Fine, now it's much easier to understand. Thank you for explanation and comments :)